src/_gettsim/ids.py

```diff
@@ -55,103 +46,53 @@ def fg_id(  # noqa: PLR0912
     Maximum of two generations, the relevant base unit for Bürgergeld / Arbeitslosengeld
     2, before excluding children who have enough income fend for themselves.
     """
-    # Build indexes
-    p_id_to_index: dict[int, int] = {}
-    p_id_to_p_ids_children: dict[int, list[int]] = {}
-
-    for index, current_p_id in enumerate(map(int, p_id)):
-        # Fast access from p_id to index
-        p_id_to_index[current_p_id] = index
-
-        # Fast access from p_id to p_ids of children
-        current_familie__p_id_elternteil_1 = int(familie__p_id_elternteil_1[index])
-        current_familie__p_id_elternteil_2 = int(familie__p_id_elternteil_2[index])
-
-        if current_familie__p_id_elternteil_1 >= 0:
-            if current_familie__p_id_elternteil_1 not in p_id_to_p_ids_children:
-                p_id_to_p_ids_children[current_familie__p_id_elternteil_1] = []
-            p_id_to_p_ids_children[current_familie__p_id_elternteil_1].append(
-                current_p_id
-            )
-
-        if current_familie__p_id_elternteil_2 >= 0:
-            if current_familie__p_id_elternteil_2 not in p_id_to_p_ids_children:
-                p_id_to_p_ids_children[current_familie__p_id_elternteil_2] = []
-            p_id_to_p_ids_children[current_familie__p_id_elternteil_2].append(
-                current_p_id
-            )
-
-    p_id_to_fg_id = {}
-    next_fg_id = 0
-
-    for index, current_p_id in enumerate(map(int, p_id)):
-        # Already assigned a fg_id to this p_id via einstandspartner /
-        # parent
-        if current_p_id in p_id_to_fg_id:
-            continue
-
-        p_id_to_fg_id[current_p_id] = next_fg_id
-
-        current_hh_id = int(hh_id[index])
-        current_p_id_einstandspartner = int(
-            arbeitslosengeld_2__p_id_einstandspartner[index]
+    n = np.max(p_id) + 1
+
+    familie__p_id_elternteil_1_loc = familie__p_id_elternteil_1
```

Comment:
```suggestion
    p_id_elternteil_1_loc = familie__p_id_elternteil_1
```
No need to carry around the namespace for local variables. 

- A couple of comments in the body of this function would help future readability
- The conditions in `np.where` for the second and third statements assigning to `fg_id` seem complicated enough to factor out into a private function?

Reply:
I added the new function and also added some comments to some of the other functions.